rotkehlchen/db/addressbook.py

```diff
@@ -179,20 +168,49 @@ def delete_addressbook_entries(
                 # each address has one or more rows
                 write_cursor.executemany(delete_without_blockchain, bindings_without_blockchain)
 
+    @overload
+    def get_addressbook_entry_name(
+            self,
+            book_type: Literal[AddressbookType.USER],
+            chain_address: OptionalChainAddress,
+    ) -> Optional[str]:
+        ...
+
+    @overload
+    def get_addressbook_entry_name(
+            self,
+            book_type: Literal[AddressbookType.GLOBAL],
+            chain_address: OptionalChainAddress,
+            priorities: dict[GlobalAddressbookSource, int],
+    ) -> Optional[str]:
+        ...
+
     def get_addressbook_entry_name(
             self,
             book_type: AddressbookType,
             chain_address: OptionalChainAddress,
+            priorities: Optional[dict[GlobalAddressbookSource, int]] = None,
     ) -> Optional[str]:
         """
         Returns the name for the specified address and blockchain or None if either there is
         no name set or the pair (address, blockchain) doesn't exist in the database.
         """
-        with self.read_ctx(book_type) as read_cursor:
-            query = read_cursor.execute(
-                'SELECT name FROM address_book WHERE address=? AND blockchain IS ?',
-                (chain_address.address, chain_address.blockchain.value if chain_address.blockchain is not None else None),  # noqa: E501
-            )
-            result = query.fetchone()
-
-        return None if result is None else result[0]
+        if book_type == AddressbookType.USER:
+            query = 'SELECT name FROM address_book WHERE address=? AND blockchain IS ?'
+        else:
+            query = 'SELECT name, source FROM address_book WHERE address=? AND blockchain IS ?'
+        bindings = [chain_address.address, chain_address.blockchain.value if chain_address.blockchain is not None else None]  # noqa: E501
+        with self.read_ctx(book_type) as cursor:
+            result = cursor.execute(query, bindings).fetchall()
+
+        if len(result) == 0:
+            return None
+        elif book_type == AddressbookType.USER:
+            return result[0][0]  # There is always a single entry for user address book
+        else:  # global address book
+            deserialized_result = [
+                (name, GlobalAddressbookSource.deserialize_from_db(source))
+                for name, source in result
+            ]
+            deserialized_result.sort(key=lambda entry: priorities[entry[1]])  # type: ignore  # for global address book priorities is not None  # noqa: E501
```

Comment:
why is the type not properly detected if you have the function overloaded? Maybe is because you need to add it as None to the other definition?